NuRadioMC/EvtGen/NuRadioProposal.py

```diff
@@ -654,7 +667,8 @@ def get_secondaries_array(self,
                         # We have already handled the muon, remove it to avoid double counting.
                         decay_products.remove(decay_particle)
 
-            grouped_decay_products = self.__group_decay_products(decay_products, min_energy_loss, lepton_position)
+            decay_energy = convert_energy_units(secondaries.final_state().energy)  # energy of the lepton before decay
```

Comment:
I did not check... but "final state" sounds suspicious? I hope you verified that this is the correct quantity?

Reply:
I have to admit I relied on what @Jean1995 told me.

Reply:
okay, I x-checked `decay_energy` with 
```
 decay_energies = np.array([p.energy for p in secondaries.decay_products()])
 decay_energy2 = np.sum(decay_energies) / pp_MeV * units.MeV
```
if the particle decays they agree perfectly, otherwise not but otherwise the variable is also not used!

Reply:
@shallmann fine?

Reply:
Yes, final_state() returns the last particle state of the propagated particle, so the energy immediately before the particle decayed (or before the propagation stopped for any other reason).